### ISSUE TITLE
Support optionally verifying SSL connection to servers

### ISSUE THREAD

**Issue by digitalcircuit** [magne4000/quassel-webserver#193]
## In brief
Quassel Webserver should allow requiring an SSL connection to servers and make verified SSL the default when the user specifically enables SSL, as per the Quassel desktop client.

Rationale, details, and more [in the original Quassel pull request #217](https://github.com/quassel/quassel/pull/217 )

## Changes required
* Implement ```Verify connection security``` checkbox
 * Grayed out (*disabled*) unless SSL is checked
 * Checked by default, unless core is too old (*possibly show a warning if this is the case*)
 * [Settings UI code, including enable checkbox](https://github.com/quassel/quassel/blob/8443ab1e0a812d76d5efa816d5bd3d476d67ab2d/src/qtui/settingspages/servereditdlg.ui#L88-L120 )
* Handle the [core ```Feature``` flag ```VerifyServerSSL = 0x0040```](https://github.com/quassel/quassel/blob/8443ab1e0a812d76d5efa816d5bd3d476d67ab2d/src/common/quassel.h#L75 )
* Add support for new [```Network::Server``` property ```sslVerify```](https://github.com/quassel/quassel/blob/master/src/common/network.cpp#L1209 )
 * [Enable ```sslVerify``` by default, even if ```useSSL``` is disabled](https://github.com/quassel/quassel/blob/master/src/common/network.h#L127-L128 )

*Possibly some other stuff..?  Not sure.*

**Comment by digitalcircuit** [comment]
Relatively small note that I forgot to mention specifically - checking ```Use secure connection``` should toggle the server port from ```6667``` → ```6697``` and vice-versa when unchecking, ```6697``` → ```6667```, **unless** the port has been manually changed.

You can see the full reasoning for this in [this commit: ```Sync default port with Use encrypted connection```](https://github.com/quassel/quassel/pull/217/commits/12d447f06dd4542698101108f1ba67ad8576de58 ).  A short demonstration is included [in the pull request](https://github.com/quassel/quassel/pull/217 ), under *Examples* → *Client Interface* with the header *Toggling 'Use encrypted connection' checkbox, add server dialog*.

In short, for those who don't know the ports but want a secure connection, this change does the right thing in most cases :)

Otherwise, in a quick test this seems to function as expected - thank you for your work!

**Comment by digitalcircuit** [comment]
Wow, that was fast.  A quick check shows it as working.  Thanks again!
